### DISCUSSION TITLE
Does SignalR client receive unsubscribed messages?

### DISCUSSION THREAD

**Discussion by Malayp** [dotnet/aspnetcore#24909] [General]
If I have a signalR hub that sends messages to all clients with a fairly large payload or private information, do all clients that connect into the hub (part of the same group) receive the messages even though they don't subscribe to those events on the client side?

**Comment by BrennanConroy** [comment]
Yes. To avoid that you can setup groups and only have the clients subscribe to a group they want to receive messages for and then send to groups on the server side instead of broadcasting to all clients.
